coderd/database/dbauthz/dbauthz.go

```diff
@@ -201,6 +201,23 @@ func ActorFromContext(ctx context.Context) (rbac.Subject, bool) {
 	return a, ok
 }
 
+type workspaceRBACContextKey struct{}
+
+// WithWorkspaceRBAC attaches a workspace RBAC object to the context.
+// RBAC fields on this RBAC object should not be used.
+//
+// This is primarily used by the workspace agent RPC handler to cache workspace
+// authorization data for the duration of an agent connection.
+func WithWorkspaceRBAC(ctx context.Context, rbacObj rbac.Object) context.Context {
+	return context.WithValue(ctx, workspaceRBACContextKey{}, rbacObj)
+}
```

Comment:
Add some guards to make sure the `rbacObj` type is `Workspace`, that ACL's are empty, that owners are set.